cardano node bump 1.15.0

paweljakubas opened the pull request:
# Issue Number

<!-- Put here a reference to the issue this PR relates to and which requirements it tackles -->


# Overview

<!-- Detail in a few bullet points the work accomplished in this PR -->

- [x] I have changed yaml
- [x] I have updated matrix
- [x] I have updated our code to make it compiling
- [ ] All test passes


# Comments

<!-- Additional comments or screenshots to attach if any -->

<!-- 
Don't forget to:

 ✓ Self-review your changes to make sure nothing unexpected slipped through
 ✓ Assign yourself to the PR
 ✓ Assign one or several reviewer(s)
 ✓ Once created, link this PR to its corresponding ticket
 ✓ Assign the PR to a corresponding milestone
 ✓ Acknowledge any changes required to the Wiki
-->

paweljakubas committed:
update dependencies

paweljakubas committed:
correct all except one things to make it compile

paweljakubas committed:
update matrix

paweljakubas committed:
update TransactionSpec shelley to new Cardano.Api changes

paweljakubas committed:
adjust compatibility and network of byron of new developmnets

paweljakubas committed:
change encoding when interacting with cardano-cli

rvl committed:
Also update niv source for cardano-node

rvl committed:
HLint

rvl committed:
Weeder

KtorZ commented on lib/shelley/src/Cardano/Wallet/Shelley/Launch.hs:
```diff
@@ -899,76 +899,76 @@ takeFaucet dir = takeMVar faucets >>= \case
+          [ "type" .= Aeson.String "Node operational certificate issue counter"
+          , "description" .= Aeson.String "Next certificate issue number: 0"
+          , "cborHex" .= Aeson.String
+              "00"
```
I'll have a look at that one :point_up:

KtorZ committed:
re-generate test credentials for the BFT-leader

KtorZ committed:
add public key to operational certificate counter

Now part of the counter for some reason?

KtorZ reviewed (approved):
:heavy_check_mark:

KtorZ committed:
increase the delta between small and big state when comparing them in test

It seems that we are getting very small rewards in CI (the root cause is unsure, but it could be that pools are punished for missing slots). A difference may not be noticeable if the stake difference is insufficient for this test. So increasing the gap even further to increase the chance of finding a difference
